lib/version.js

```diff
@@ -15,7 +15,7 @@ function latest(versions) {
   let version = '';
   let origVersions = versions;
   versions = versions.filter(function(version) {
-    return (/^v?[0-9]/).test(version);
+    return (/v?[0-9]/).test(version);
   });
   try {
```

Comment:
I wonder if it would be more robust if we tried to find a dotted-numeric or semver substring, and passing that to the various sort routines?

[This library](https://github.com/ragingwind/semver-sort/blob/master/index.js) uses a semver regex (which we have as a dev dependency). Though probably we should probably do something a little different so we don't erroneously strip out the last part of e.g. `1.0.0.3`.

Here's an idea:

1. As before, discard anything without a number.
1. First, attempt to preprocess the strings:
    - If a string is a prefix followed by a valid semver, and that's it, keep the semver part.
    - Else if the string is a prefix followed by dotted numbers followed by anything else, keep the dotted numbers.
    - Discard any other strings else (to avoid classifying as latest some random tag like `orig-1` among a bunch of version numbers).
2. If the preprocessing step didn't yield any strings, fall back to sorting all of the original strings lexicographically.
3. If all the preprocessed strings are valid semver, sort them using semver.
4. If not, sort them using latest dotted version.